src/services/navigationBar.ts

```diff
@@ -192,6 +193,11 @@ namespace ts.NavigationBar {
                             addTopLevelNodes((<Block>functionDeclaration.body).statements, topLevelNodes);
                         }
                         break;
+
+                    default:
+                        const childrens: Node[] = [];
+                        forEachChild(node, child => { childrens.push(child) });

```

Comment:
We should change this function to take a node instead of a list. the allocations here are not really needed. we put nodes in a new array just to call the function, then throw them away.